lib/dbservice_web/controllers/student_controller.ex

```diff
@@ -562,4 +573,89 @@ defmodule DbserviceWeb.StudentController do
       _ -> true
     end
   end
+
+  # function to verify student data with the values recieved in the verification params
+  swagger_path :verify_student do
+    post("/api/student/verify-student")
+
+    parameters do
+      body(:body, Schema.ref(:VerifyStudentRequest), "The verify student request", required: true)
+    end
+
+    response(200, "OK", Schema.ref(:VerificationResult))
+  end
+
+  def verify_student(conn, %{
+        "student_id" => student_id,
+        "verification_params" => verification_params
+      }) do
+    case get_student_and_user(student_id) do
+      {:ok, student, user} ->
+        student_exists = verify_student_and_user_data(student, user, verification_params)
+
+        conn
+        |> put_status(:ok)
+        |> json(%{is_verified: student_exists})
+
+      {:error, :not_found} ->
+        conn
+        |> put_status(:not_found)
+        |> json(%{error: "Student not found"})
+    end
+  end
+
+  defp get_student_and_user(student_id) do
+    case Users.get_student_by_student_id(student_id) do
+      nil ->
+        {:error, :not_found}
+
+      student ->
+        case Users.get_user!(student.user_id) do
+          nil ->
+            {:error, :not_found}
+
+          user ->
+            {:ok, student, user}
+        end
+    end
+  end
+
+  defp verify_student_and_user_data(student, user, verification_params) do
+    Enum.all?(verification_params, fn {key, value} ->
+      cond do
+        key == "auth_group_id" ->
+          verify_auth_group(user.id, value)
+
+        key == "date_of_birth" ->
+          user_dob = Map.get(user, String.to_existing_atom(key))
+          parsed_value = Date.from_iso8601!(value)
+          Date.compare(user_dob, parsed_value) == :eq
```

Comment:
What happens if `user_dob` and `parsed_value` are not equal?

Reply:
It returns false if user_dob and parsed_value are equal, and if any condition is false then the function inside Enum.all() will return false.